New data source `azurerm_search_service` (#10181)
terraform-providers/terraform-provider-azurerm: website/docs/d/search_service.html.markdown

layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_search_service"
description: |-
  Manages a Search Service.
---

# azurerm_search_service

Manages a Search Service.

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

data "azurerm_search_service" "example" {
  name                = "example-search-service"
  resource_group_name = azurerm_resource_group.example.name
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The Name which should be used for this Search Service. Changing this forces a new Search Service to be created.

* `resource_group_name` - (Required) The name of the Resource Group where the Search Service should exist. Changing this forces a new Search Service to be created.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported:

* `id` - The ID of the Search Service.

* `primary_key` - The Primary Key used for Search Service Administration.

* `secondary_key` - The Secondary Key used for Search Service Administration.

* `query_keys` - A `query_keys` block as defined below.

> * `public_network_access_enabled` - Whether or not public network access is allowed for this resource. Defaults to `true`.

Reviewer: this is a Data Source which is read-only, so the default value doesn't apply here - could we also update the messaging to highlight this isn't configurable:

 ```suggestion
* `public_network_access_enabled` - Whether or not public network access is enabled for this Search Service.
```

* `partition_count` - The number of partitions which have been created.

* `replica_count` - The number of replica's which have been created.

> -> **Note:** `partition_count` and `replica_count` can only be configured when using a `standard` sku.

Reviewer: since this is a data source, this isn't configurable - so we can remove this (since the fields should get a default value either way)

```suggestion
```

* `identity` - An `identity` block as defined below.

> * `tags` - (Optional) A mapping of tags which should be assigned to the Search Service.

Reviewer: (as above) since this isn't configurable can we update the wording to highlight that this is read only:

```suggestion
* `tags` - A mapping of tags which are assigned to the Search Service.
```

Author: We don't actually set the tags on this data source so I'll just remove this line 😄

---

A `identity` block supports the following:

> * `type` - The Type of Identity which should be used for the Search Service. At this time the only possible value is `SystemAssigned`.

Reviewer: (as above)

```suggestion
* `type` - The type of Managed Identity which used for the Search Service.
```
